apps/ledger-live-mobile/src/newArch/features/FirmwareUpdate/components/UpdateBanner/index.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { memo } from "react";
+import { Pressable } from "react-native";
+import { useTranslation } from "react-i18next";
+import { Text, Flex, Icons } from "@ledgerhq/native-ui";
+import { DownloadMedium, UsbMedium } from "@ledgerhq/native-ui/assets/icons";
+import { DeviceModelId, getDeviceModel } from "@ledgerhq/devices";
+
+import Button from "../../../../../components/Button";
+import QueuedDrawer from "../../../../../components/QueuedDrawer";
+import { UpdateStep } from "~/screens/FirmwareUpdate";
+import { useUpdateBannerViewModel } from "./useUpdateBannerViewModel";
+import { ViewProps } from "./ViewProps";
+
+export type FirmwareUpdateBannerProps = {
+  onBackFromUpdate: (updateState: UpdateStep) => void;
+};
+
+const UpdateBanner = ({
```

Comment:
I'm not sure why the component and the export are defined separately. The props passed to the component is the returns of a hook, you can put it in the component with a destruction of the return value.
```TypeScript
export default memo(function(outerProps: FirmwareUpdateBannerProps ) {
    const { ... }: ViewProps = useUpdateBannerViewModel(outerProps);
    return <Component />;
});
```

Reply:
This is [the new convention](https://ledgerhq.atlassian.net/wiki/spaces/WXP/pages/4508058220/Code+Guidelines+-+React+Native#useComponentViewModel) that was proposed by @KVNLS last week.
It allows to decouple the rendering logic and the business model. But tell me maybe I don't understand exactly your point.

Reply:
Ok, now that it's in the convention, following it is good.